bindings: ethernet: replace phy-dev with phy-handle

fabiobaltieri opened the pull request:
Rename the phy-dev property with phy-handle to match the Linux ethernet-controller binding and move it up to ethernet.yaml so that it can be used by other drivers.

---

This is currently only used in one driver but I'll move the other over, that way we can check the init priority as well.

Reference for the change:

https://elixir.bootlin.com/linux/v6.5-rc3/source/Documentation/devicetree/bindings/net/ethernet-controller.yaml#L122

https://github.com/zephyrproject-rtos/zephyr/pull/60816#discussion_r1276175195

fabiobaltieri committed:
bindings: ethernet: replace phy-dev with phy-handle

Rename the phy-dev property with phy-handle to match the Linux
ethernet-controller binding and move it up to ethernet.yaml so that it
can be used by other drivers.

Signed-off-by: Fabio Baltieri <fabiobaltieri@google.com>

manuargue reviewed (dismissed):
LGTM

fabiobaltieri commented:
> @fabiobaltieri about the sample mentioned in [#58388 (comment)](https://github.com/zephyrproject-rtos/zephyr/pull/58388#issue-1730503586) for this driver. What would be the updates/test needed? maybe I can help on that

Hi @manuargue, I'll write an update there, that's the latest I need to chase down to have that passing in CI :-)

fabiobaltieri commented:
update: set back `phy-handle` as required in the nxp binding

nandojve commented on dts/bindings/ethernet/ethernet.yaml:
```diff
@@ -21,3 +22,8 @@ properties:
 
       If set we ignore any setting of the local-mac-address property.
+
+  phy-handle:
```
Non blocking. I still thinking that this is a good opportunity to rename the file `ethernet.yaml` to `ethernet-controller.yaml` as a preparation to property rename. Otherwise project will be always in debt and structural changes in future could be more difficult since code base is growing.

manuargue replied:
Hi @nandojve I’m curious, is there any umbrella issue where property rename to follow Linux bindings is being discussed?

nandojve replied:
Hi @manuargue , not that I have knowledge. The policy in general for networking is being close to Linux. Zephyr has their own autonomy but in this case being equal in the filename and property makes sense. This changed was motivated by #60816.

fabiobaltieri replied:
Cool with that, added a patch to rename the file.

fabiobaltieri committed:
bindings: ethernet: rename ethernet to ethernet-controller

Rename the ethernet.yaml template to ethernet-controller.yaml to match
the Linux one.

Signed-off-by: Fabio Baltieri <fabiobaltieri@google.com>
